index.d.ts

```diff
@@ -468,10 +469,56 @@ interface ConfigParams {
   transactionCookie?: Pick<CookieConfigParams, 'sameSite'>;
 
   /**
-   * String value for the client's authentication method. Default is `none` when using response_type='id_token', otherwise `client_secret_basic`.
+   * String value for the client's authentication method. Default is `none` when using response_type='id_token', `private_key_jwt` when using a `clientAssertionSigningKey`, otherwise `client_secret_basic`.
    */
   clientAuthMethod?: string;
 
+  /**
+   * Private key for use with 'private_key_jwt' clients.
+   *
+   * Can be a PEM:
+   *
+   * ```js
+   * app.use(auth({
+   *   ...
+   *   clientAssertionSigningKey: '-----BEGIN PRIVATE KEY-----\nMIIEo...PgCaw\n-----END PRIVATE KEY-----',
+   * }))
+   * ```
+   *
+   * Or JWK:
+   *
+   * ```js
+   * app.use(auth({
+   *   ...
+   *   clientAssertionSigningKey: {
+   *     kty: 'RSA',
+   *     e: 'AQAB',
+   *     n: '12oBZRhCiZFJLcPg59LkZZ9mdhSMTKAQZYq32k_ti5SBB6jerkh-WzOMAO664r_qyLkqHUSp3u5SbXtseZEpN3XPWGKSxjsy-1JyEFTdLSYe6f9gfrmxkUF_7DTpq0gn6rntP05g2-wFW50YO7mosfdslfrTJYWHFhJALabAeYirYD7-9kqq9ebfFMF4sRRELbv9oi36As6Q9B3Qb5_C1rAzqfao_PCsf9EPsTZsVVVkA5qoIAr47lo1ipfiBPxUCCNSdvkmDTYgvvRm6ZoMjFbvOtgyts55fXKdMWv7I9HMD5HwE9uW839PWA514qhbcIsXEYSFMPMV6fnlsiZvQQ',
+   *   },
+   * }))
+   * ```
+   */
+  clientAssertionSigningKey?: KeyInput | object;
```

Comment:
[KeyObject](https://nodejs.org/api/crypto.html#class-keyobject) should be fine too, no? And we should probably type JWK better than `object`, this version of `jose` exports `JSONWebKey`

Reply:
👍  added `KeyObject` and `JSONWebKey`